docs/blogs/zoda.md

```diff
@@ -0,0 +1,372 @@
+---
+title: "Optimizing Block Dispersal with Coding"
+description: "In which we describe how we use coding schemes, and ZODA to distribute blocks efficiently"
+date: "October 15th, 2025"
+author: "Lucas Meier"
+author_twitter: "https://x.com/cronokirby"
+katex: true
+---
+
+A crucial part of blockchain consensus is distributing the block among participants.
+You can come to consensus over a mere fingerprint of the block---a hash for example---
+but doing anything interesting with the block, like processing transactions, requires
+the data.
+
+We've invested some effort in this part of the Commonware stack recently, and I'd
+like to share some fruits of that effort in this post.
+As an outline, we'll cover:
+
+- data dissemination, naively,
+- how to increase efficiency with coding,
+- and, finally, how to get quicker guarantees about the data with [ZODA](https://eprint.iacr.org/2025/034).
+
+In a future post we could cover lower-level details of [our implementation](https://github.com/commonwarexyz/monorepo/blob/main/coding/src/zoda.rs), like
+the field we use for Reed-Solomon coding, and the optimizations needed for
+fast fourier transforms, but this one will stick to an overview.
+
+## Naive Dissemination
+
+We assume a _leader_ has some data, $D$ bytes worth, to be disseminated.
+They want $m$ followers to receive it.
+The simplest approach is to have the leader send the data to every follower.
+The leader's transmission cost is $m \cdot D$ bytes, and the followers' cost
+is $0$ bytes, since they send nothing.
+
+![](/blogs/zoda/img-000.png)
+
+Networked protocols are often bottlenecked by sending data, since moving
+bits around the planet, a country, or a building is hopelessly slow compared
+to moving it within an integrated circuit.
+
+Some back-of-the-napkin math:
+if our leader has a 1 Gb / s link, and needs to send a 1 MB block to 50 followers,
+we would expect this to take at least 400 ms, assuming no overhead at all
+in transmitting.
+Our [estimator crate](https://docs.rs/commonware-estimator/latest/estimator/)
+
+This protocol is bottlenecked more so, with all of its communication going through
+the single leader node.
+While the leader sends the entire data $m$ times, the followers sit idle,
+wasting their resources.
+We can do better.
```

Comment:
nit: may want to link to ordered broadcast as a way to do better with DSMR

Reply:
I'm not quite sure what that reference would be?